pkg/catalog/catalog.go

```diff
@@ -102,9 +104,10 @@ func getRegistryDevfiles(registry Registry) ([]DevfileComponentType, error) {
 	}
 	registry.URL = URL
 	indexLink := registry.URL + indexPath
-	jsonBytes, err := util.HTTPGetRequest(indexLink)
+	token, _ := keyring.Get(credentialPrefix+registry.Name, "default")
```

Comment:
It's pretty hard to differentiate as the only difference between secured and unsecured registries is whether users pass token or not. Given odo doesn't have a way to know if the registry is secured or not (only users know their registry type and odo just helps users to access the registry), so I think the question can be boiled down to "how does odo knows what time is necessary to call `keyring.Get` and return proper error". The problem is that _using token or not_ is the only indicator that we can use to answer the question, but we only know _using token or not_ when users run `odo registry` command, we don't know _using token or not_ when users use `odo create` or `odo catalog` to access the registry. I think the only way we can do is to save _using token or not_ information in memory, but that's also not ideal as we expose security related information.

Regarding the error handling
> If there's any issues with the user's keyring setup for secured registries, we won't get an error

This error can be detected when user adds the registry unless there is a edge case where user changes the keyring setup after adding the registry and before using the registry

> If there's any issues retrieving the token for a secured registry, we won't get an error

We won't missing the error but the error is not that specific because we can catch the error that secured registry can't not be accessed and let user check the token/keyring

Reply:
If a user adds a secured registry with `odo registry add —token`, couldn’t we store whether or not it’s secured (maybe a `secured: true` in env.yaml)?

Reply:
If the users adds a secured registry to odo with `odo registry add —token`, couldn’t we store whether or not it’s secured with `secured: true` in the env.yaml? Then when we go to list the devfiles in the registry, we’d easily know whether or not we need to retrieve a token

Reply:
Since the token would still be stored securely in the keychain, that should be fine from a security perspective, shouldn’t it? I don’t think mentioning whether or not a specific registry requires a token would make it more insecure, in my opinion.

Reply:
My original concern is we should hide all the security related information for user, but I think it again, it not bad to display whether or not a specific registry is secure for user, so the changes will be in my next commit.